File: sdks/php/rest/src/Types/ProjectResponse.php
<?php

namespace Brume\Rest\Types;

use Brume\Rest\Core\Json\JsonSerializableType;
use Brume\Rest\Core\Json\JsonProperty;

/**
 * `GET /v1/project` success response.
 */
class ProjectResponse extends JsonSerializableType
{
    /**
     * @var string $createdAt RFC-3339 timestamp of project creation.
     */
    #[JsonProperty('created_at')]
    public string $createdAt;

    /**
     * @var string $id Hex-encoded 16-byte project id.
     */
    #[JsonProperty('id')]
    public string $id;

    /**
     * @var int $maxConnections
     */
    #[JsonProperty('max_connections')]
    public int $maxConnections;

    /**
     * @var string $name
     */
    #[JsonProperty('name')]
    public string $name;

    /**
     * @var string $tier
     */
    #[JsonProperty('tier')]
    public string $tier;

    /**
     * @param array{
     *   createdAt: string,
     *   id: string,
     *   maxConnections: int,
     *   name: string,
     *   tier: string,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->createdAt = $values['createdAt'];
        $this->id = $values['id'];
        $this->maxConnections = $values['maxConnections'];
        $this->name = $values['name'];
        $this->tier = $values['tier'];
    }

    /**
     * @return string
     */
    public function __toString(): string
    {
        return $this->toJson();
    }
}

File: sdks/csharp/rest/src/Brume.Rest/Types/ProjectResponse.cs
using Brume.Rest.Core;
using global::System.Text.Json;
using global::System.Text.Json.Serialization;

namespace Brume.Rest;

/// <summary>
/// `GET /v1/project` success response.
/// </summary>
[Serializable]
public record ProjectResponse : IJsonOnDeserialized
{
    [JsonExtensionData]
    private readonly IDictionary<string, JsonElement> _extensionData =
        new Dictionary<string, JsonElement>();

    /// <summary>
    /// RFC-3339 timestamp of project creation.
    /// </summary>
    [JsonPropertyName("created_at")]
    public required string CreatedAt { get; set; }

    /// <summary>
    /// Hex-encoded 16-byte project id.
    /// </summary>
    [JsonPropertyName("id")]
    public required string Id { get; set; }

    [JsonPropertyName("max_connections")]
    public required long MaxConnections { get; set; }

    [JsonPropertyName("name")]
    public required string Name { get; set; }

    [JsonPropertyName("tier")]
    public required string Tier { get; set; }

    [JsonIgnore]
    public ReadOnlyAdditionalProperties AdditionalProperties { get; private set; } = new();

    void IJsonOnDeserialized.OnDeserialized() =>
        AdditionalProperties.CopyFromExtensionData(_extensionData);

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
